src/GitHubExtension/DataModel/DataObjects/Release.cs

```diff
@@ -0,0 +1,155 @@
+// Licensed under the MIT License.
+
+using Dapper;
+using Dapper.Contrib.Extensions;
+using GitHubExtension.Helpers;
+
+namespace GitHubExtension.DataModel;
+
+[Table("Release")]
+public class Release
+{
+    [Key]
+    public long Id { get; set; } = DataStore.NoForeignKey;
+
+    public long InternalId { get; set; } = DataStore.NoForeignKey;
+
+    // Repository table
+    public long RepositoryId { get; set; } = DataStore.NoForeignKey;
+
+    public string Name { get; set; } = string.Empty;
+
+    public string TagName { get; set; } = string.Empty;
+
+    public long Prerelease { get; set; } = DataStore.NoForeignKey;
+
+    public string HtmlUrl { get; set; } = string.Empty;
+
+    public long TimeCreated { get; set; } = DataStore.NoForeignKey;
+
+    public long TimePublished { get; set; } = DataStore.NoForeignKey;
+
+    public long TimeLastObserved { get; set; } = DataStore.NoForeignKey;
+
+    [Write(false)]
+    private DataStore? DataStore
+    {
+        get; set;
+    }
+
+    [Write(false)]
+    [Computed]
+    public DateTime CreatedAt => TimeCreated.ToDateTime();
+
+    [Write(false)]
+    [Computed]
+    public DateTime? PublishedAt => TimePublished != 0 ? TimePublished.ToDateTime() : null;
+
+    [Write(false)]
+    [Computed]
+    public DateTime LastObservedAt => TimeLastObserved.ToDateTime();
+
+    public override string ToString() => Name;
+
+    public static Release GetOrCreateByOctokitRelease(DataStore dataStore, Octokit.Release okitRelease, long repositoryId)
+    {
+        var release = CreateFromOctokitRelease(dataStore, okitRelease, repositoryId);
+        return AddOrUpdateRelease(dataStore, release);
+    }
+
+    public static IEnumerable<Release> GetAllForRepository(DataStore dataStore, Repository repository)
+    {
+        var sql = $"SELECT * FROM Release WHERE RepositoryId = @RepositoryId ORDER BY TimePublished DESC;";
+        var param = new
+        {
+            RepositoryId = repository.Id,
+        };
+
+        Log.Logger()?.ReportDebug(DataStore.GetSqlLogMessage(sql, param));
+        var releases = dataStore.Connection!.Query<Release>(sql, param, null) ?? Enumerable.Empty<Release>();
+        foreach (var release in releases)
+        {
+            release.DataStore = dataStore;
+        }
+
+        return releases;
+    }
+
+    public static Release? GetByInternalId(DataStore dataStore, long internalId)
+    {
+        var sql = $"SELECT * FROM Release WHERE InternalId = @InternalId;";
+        var param = new
+        {
+            InternalId = internalId,
+        };
+
+        var release = dataStore.Connection!.QueryFirstOrDefault<Release>(sql, param, null);
+        if (release is not null)
+        {
+            // Add Datastore so this object can make internal queries.
+            release.DataStore = dataStore;
+        }
+
+        return release;
+    }
+
+    public static void DeleteLastObservedBefore(DataStore dataStore, long repositoryId, DateTime date)
```

Comment:
Need another Delete for obsolete data. This should be any data from any repository that has not been observed recently (a few days is reasonable), so any widgets that are removed will get cleaned up from the datastore after some time.